src-built-in/components/appMarket/src/stores/storeActions.js

```diff
@@ -0,0 +1,288 @@
+        value: []
+    });
+}
+
+/**
+ * Async function to fetch apps from the FDC3 api (appD)
+ */
+function getApps() {
+    appd.getAll((err, apps) => {
+        getStore().setValue({
+            field: 'installed',
+            value: [apps[0].appId]
+        });
+    });
+    return appd.getAll();
+}
+
+/**
+ * Function to "install" an app. Adds the id to a list of installed apps
+ * @param {string} name The name of the app
+ */
+function addApp(name) {
+    let apps = getStore().getValue({
+        field: 'apps'
+    });
+
+    let installed = getStore().getValue({
+        field: 'installed'
+    });
+
+    for (let i = 0; i < apps.length; i++) {
+        let app = apps[i];
+        let thisAppName = app.title ? app.title : app.name;
+        if (thisAppName === name && !installed.includes(app.appId)) {
+            installed.push(app.appId);
+            break;
+        }
+    };
+
+    getStore().setValue({
+        field: 'installed',
+        value: installed
+    });
+}
+
+/**
+ * Function to "uninstall" an app. Removes the id from a list of installed apps
+ * @param {string} name The name of the app
+ */
+function removeApp(name) {
+    let apps = getStore().getValue({
+        field: 'apps'
+    });
+
+    let installed = getStore().getValue({
+        field: 'installed'
+    });
+
+    for (let i = 0; i < apps.length; i++) {
+        let app = apps[i];
+        let thisAppName = app.title ? app.title : app.name;
+        if (thisAppName === name && installed.includes(app.appId)) {
+            let index = installed.indexOf(app.appId);
+            installed.splice(index, 1);
+            break;
+        }
+    }
+
+    getStore().setValue({
+        field: 'installed',
+        value: installed
+    });
+}
+
+/**
+ * Gets the list of installed apps
+ */
+function getInstalledApps() {
+    return getStore().getValue({
+        field: 'installed'
+    });
+}
+
+/**
+ * Gets the list of filtered apps (when searching/filtering by tags)
+ */
+function getFilteredApps() {
+    return getStore().getValue({
+        field: 'filteredApps'
+    });
+}
+
+/**
+ * Clears the list of filtered apps
+ */
+function clearFilteredApps() {
+    getStore().setValue({
+        field: 'filteredApps',
+        value: []
+    });
+}
+
+/**
+ * Async call to appD to get the list of all tags
+ */
+function getTags() {
+    return appd.getTags()
+}
+
+/**
+ * Gets the list of active tags (these are tags that are actively filtering the content list)
+ */
+function getActiveTags() {
+    return getStore().getValue({
+        field: 'activeTags'
+    });
+}
+
+/**
+ * Adds an 'active tag' to the list of filtered tags
+ * @param {string} tag The tag name
+ */
+function addTag(tag) {
+    _addActiveTag(tag);
+}
+
+/**
+ * Removes an 'active tag' from the list of filtered tags
+ * @param {string} tag The tag name
+ */
+function removeTag(tag) {
+    _removeActiveTag(tag);
+}
+
+/**
+ * Removes all tags from the active tags list
+ */
+function clearTags() {
```

Comment:
Not fully sure of the purpose of this funciton. And the ones above it. Just having public functions in case you need to mutate input in the future?

Reply:
I definitely use these functions. Could just expose the private versions but wanted cleaner names for the versions the components are calling